src/App.tsx

```
@@ -1,60 +1,162 @@
+  const [isCommentsError, setIsCommentsError] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  console.log(comments);
 
-              <Loader />
+  useEffect(() => {
+    client.get<User[]>('/users').then(setUsers);
+  }, []);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  const handleActiveUser = (user: User) => {
+    setActiveUser(null);
+    setActivePost(null);
+    setIsLoading(true);
+    setIsError(false);
+
+    setActiveUser(user);
+    client
+      .get<Post[]>(`/posts?userId=${user?.id}`)
+      .then(setPosts)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  };
+
+  const handleActivePost = (post: Post | null) => {
+    setIsCommentsLoading(true);
+    setIsCommentsError(false);
+
+    setActivePost(post);
+    if (post !== null) {
+      client
+        .get<Comment[]>(`/comments?postId=${post?.id}`)
+        .then(setComments)
+        .catch(() => setIsCommentsError(true))
+        .finally(() => setIsCommentsLoading(false));
+    }
+  };
+
+  const handleAddComment = (newComment: Comment, postId: number) => {
+    const comment = { ...newComment, postId: postId };
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+    return client
+      .post<Comment>('/comments', comment)
+      .then(commentFromServer =>
+        setComments(prev => [...(prev || []), commentFromServer]),
+      )
+      .catch()
+      .finally();
+  };
+
+  const handleDeleteComment = (commentId: number) => {
+    client
+      .delete(`/comments/${commentId}`)
+      .then(() =>
+        setComments(prevComments =>
+          prevComments.filter(prevComment => prevComment.id !== commentId),
+        ),
+      )
+      .catch()
```

The catch block is empty. It's important to handle errors properly to avoid silent failures. Consider logging the error or providing user feedback.